packages/circuit-ui/components/Button/Button.stories.tsx

```diff
@@ -107,24 +107,30 @@ export const Tracking = (args: ButtonProps) => (
   </Button>
 );
 
-export const Loading = (args: ButtonProps) => {
-  const [loading, setLoading] = useState(false);
+const ButtonWithLoadingState = (args: ButtonProps) => {
+  const [isLoading, setLoading] = useState(false);
 
   const handleClick = () => {
     setLoading(true);
     setTimeout(() => {
       setLoading(false);
-    }, 2000);
+    }, 3000);
   };
 
   return (
     <Button
-      isLoading={loading}
+      isLoading={isLoading}
       loadingLabel="Loading"
       onClick={handleClick}
       {...args}
-    >
-      Things take time
-    </Button>
+    />
   );
 };
+
+export const Loading = (args: ButtonProps) => (
+  <ButtonWithLoadingState>{args.children}</ButtonWithLoadingState>
```

Comment:
Would it make sense to forward all props here, to try out the loading state with other configurations (e.g. `primary`, `secondary`, and `tertiary` variants)?

```suggestion
  <ButtonWithLoadingState {...args} />
```

Reply:
Ah yes, that's what I intended to do 🤦 the stateful component is already accepting it